### PR TITLE
tests(core): add ibc memo type snapshot tests

### PR DESCRIPTION
**Pull request by noot**
## Summary
follow up to #1202  - add snapshot tests for the ibc memo types.

## Background
these are protocol-breaking if they change.

## Changes
- add ibc memo type snapshot tests

### CODE REVIEW

File: crates/astria-core/Cargo.toml [hunk 1]
```diff
@@ -59,5 +59,6 @@ brotli = ["dep:brotli"]
 [dev-dependencies]
 insta = { workspace = true, features = ["json"] }
 rand = { workspace = true }
+serde_json = { workspace = true }
```
Comment (SuperFluffy) [inline crates/astria-core/Cargo.toml:62]: remove

File: crates/astria-core/src/bridge.rs [hunk 2]
```diff
@@ -35,3 +35,34 @@ pub struct Ics20TransferDepositMemo {
...
+            bridge_address: Address::builder()
+                .array([99; 20])
+                .prefix(ASTRIA_ADDRESS_PREFIX)
+                .try_build()
+                .unwrap(),
+            block_number: 1,
+            transaction_hash: [88; 32],
+        };
+
+        insta::assert_json_snapshot!(serde_json::to_string(&memo).unwrap());
```
Comment (SuperFluffy) [inline crates/astria-core/src/bridge.rs:57]: remove the string construction - `assert_json_snapshot` does that for you

File: crates/astria-core/src/bridge.rs [hunk 3]
```diff
@@ -35,3 +35,34 @@ pub struct Ics20TransferDepositMemo {
...
+    }
+
+    #[test]
+    fn ics20_transfer_deposit_memo_snapshot() {
+        let memo = Ics20TransferDepositMemo {
+            rollup_address: "some_rollup_address".to_string(),
+        };
+
+        insta::assert_json_snapshot!(serde_json::to_string(&memo).unwrap());
```
Comment (SuperFluffy) [inline crates/astria-core/src/bridge.rs:66]: as above, just pass in `memo` without `serde_json`

**Review by SuperFluffy — CHANGES_REQUESTED** [review]

File: crates/astria-core/src/bridge.rs (hunk 3 above)
Comment (noot) [reply crates/astria-core/src/bridge.rs]: done

**Review by SuperFluffy — APPROVED** [review]
Looks great, thanks!

**Pull request merged**